asr1k_neutron_l3/plugins/db/asr1k_db.py

```diff
@@ -95,6 +96,15 @@ def get_bgpvpn_advertise_extra_routes_by_router_id(self, context, router_id):
                 return True
         return False
 
+    def get_port_count_in_segment(self, context: n_context.Context, network_id: str) -> Dict[str, int]:
+        query = context.session.query(asr1k_models.ASR1KExtraAttsModel.agent_host,
+                                      func.count(asr1k_models.ASR1KExtraAttsModel.port_id).label('port_count')) \
+            .join(segment_models.NetworkSegment,
+                  segment_models.NetworkSegment.id == asr1k_models.ASR1KExtraAttsModel.segment_id) \
+            .filter(segment_models.NetworkSegment.network_id == network_id) \
+            .group_by(segment_models.NetworkSegment.id, asr1k_models.ASR1KExtraAttsModel.agent_host)
```

Comment:
Back when I wrote that, my mind thought that in theory we could have multiple physnets to the same agent. Like one on Port-channel1 and another one on Port-channel2. In that case that grouping would make sense. Right?

Reply:
physnets on port-channels don't make sense to me, I think these objects are in different domains.

In _theory_ each agent should be only mapped to one hostgroup in our ml2 driver, meaning each agent should have - with a consistent db - only one segment should exists per agent per network, everything else should be a misconfiguration on our part, I think.
If we have the case that this happens that we have two segments for the same network mapped to the same agent, then the query would return two lines for one agent, e.g. `(agent1, seg1, count1), (agent1, seg2, count2)`. In the dictionary, this would result in two cases for agent1 and one would overwrite the other like this:
```
>>> {x: n for n, x in enumerate((1,2,1,3,1))}
{1: 4, 2: 1, 3: 3}
```

Reply:
Okay, I see that possibility. The query is adjusted and does not rely on segments any more.